src/inngest/functions/create-course/index.ts

```diff
@@ -89,22 +90,36 @@ export let createCourse = inngest.createFunction(
       return await createCourseInRails(sanityBody)
     })
 
-    await step.run('add-lessons-to-course', async () => {
-      return await addLessonsToCourse(
-        sanityBody.lessonIds,
-        courseObject.data.id,
-      )
-    })
+    // await step.run('add-lessons-to-course', async () => {
+    //   return await addLessonsToCourse(
+    //     sanityBody.lessonIds,
+    //     courseObject.data.id,
+    //   )
+    // })
 
-    await step.run('update-owner-to-instructor', async () => {
-      return await updateOwnerToInstructor(
-        sanityBody.instructor.eggheadInstructorId,
-        courseObject.data.id,
-      )
-    })
+    // await step.run('update-owner-to-instructor', async () => {
+    //   return await updateOwnerToInstructor(
+    //     sanityBody.instructor.eggheadInstructorId,
+    //     courseObject.data.id,
+    //   )
+    // })
 
     await step.run('add-rails-id-to-sanity', async () => {
       return await saveCourseDataToSanity(sanityBody, courseObject.data)
     })
+
+    await step.run('upsert-course-to-typesense', async () => {
+      if (sanityBody.searchIndexingState !== 'indexed') {
+        let message = `Course is not indexed, skipping upsert to Typesense. Course: ${sanityBody.title}`
+        console.log(message)
+        return message
+      } else if (sanityBody.productionProcessState !== 'published') {
+        let message = `Course is not published, skipping upsert to Typesense. Course: ${sanityBody.title}`
+        console.log(message)
+        return message
+      } else {
+        return await upsertCourseToTypesense(courseObject.data)
+      }
```

Comment:
Since this is only being run on the initial publish of the Sanity document, I'm checking for the indexed and published attributes on the document instead of deleting any existing documents in TypeSense since there won't be any.

Reply:
deleting is a safe operation regardless. might make sense to have a `syncWithTypeSense` inngest function that can be run anytime/anywhere that handles removing and replacing.

you could call it here with `await step.invoke` ([docs for invoke](https://www.inngest.com/docs/reference/functions/step-invoke)) as a for instance

Reply:
I'd probably, generally, use the `lesson` as the source of truth for all things display etc versus using sanity fields for the current state. mostly for consistency